torch/csrc/autograd/python_function.cpp

```diff
@@ -681,6 +682,11 @@ PyObject *THPFunction_apply(PyObject *cls, PyObject *inputs)
   if (!ctx_obj) return nullptr;
   THPFunction* ctx = (THPFunction*)ctx_obj.get();
 
+  const auto& functorch_tls = at::functorch::functorchTLSAccessor();
```

Comment:
Can you move this up next to the record function just so that it is not in the middle of the ctx object initialization code?
Also add a comment saying that this is a temporary hack to improve functorch UX.

Reply:
Yup, can do